libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp

```diff
@@ -65,16 +64,31 @@ bool SyncingState::is_actively_syncing() const {
 
 void SyncingState::set_peer_malicious(const std::optional<dev::p2p::NodeID> &peer_id) {
   if (peer_id.has_value()) {
-    malicious_peers_.insert(peer_id.value());
+    malicious_peers_[peer_id.value()] = std::chrono::steady_clock::now();
     return;
   }
 
   // this lock is for peer_id_ not the malicious_peers_
   std::shared_lock lock(peer_mutex_);
-  malicious_peers_.insert(peer_->getId());
+  malicious_peers_[peer_->getId()] = std::chrono::steady_clock::now();
 }
 
-bool SyncingState::is_peer_malicious(const dev::p2p::NodeID &peer_id) const { return malicious_peers_.count(peer_id); }
+bool SyncingState::is_peer_malicious(const dev::p2p::NodeID &peer_id) {
+  if (conf_.disable_peer_blacklist) {
+    return false;
+  }
+
+  // Peers are marked malicious for the time defined in conf_.network_peer_blacklist_timeout
+  if (auto i = malicious_peers_.find(peer_id); i != malicious_peers_.end()) {
+    if (std::chrono::duration_cast<std::chrono::seconds>(std::chrono::steady_clock::now() - i->second).count() <=
+        conf_.network_peer_blacklist_timeout) {
+      return true;
+    } else {
+      malicious_peers_.erase(i);
```

Comment:
is_peer_maliciou is called from TaraxaCapability::onConnect. Are we certain that onConnect si 100% synchronous ? If not, we need mutex here

Reply:
malicious peers is a thread safe map now